### ISSUE TITLE
getting NaN on minimum price or maximum

### ISSUE THREAD

**Issue by ahmed-bhs** [angular-slider/ng5-slider#54]
I'm always getting NaN on minimum price and maximum, If the tow value s comes from a HTTP response. Maybe I should show the value asynchronsly any way.
<img src="https://cdn.discordapp.com/attachments/239433691883896833/528590736245915659/Capture_decran_de_2018-12-29_16-10-19.png">

This my component:
```
export class PriceFilterComponent implements OnChanges {
    @Input() floor;
    @Input() ceil;
    @Output() eventEmitterFilter = new EventEmitter<{}>();
    min: any;
    max: any;

    options: Options;

    ngOnChanges() {
        console.log(this.max); //undifined then NaN
        if (this.floor && this.ceil) {
            console.log(this.max); // NAN
            this.max = Number(this.ceil.toFixed(0)) || 0;
            this.min = Number(this.floor.toFixed(0)) || 0;

            this.options = {
                floor: this.floor,
                ceil: this.ceil,
                showSelectionBar: true,
                selectionBarGradient: {
                    from: 'white',
                    to: '#007bff'
                },
                showOuterSelectionBars: true,
                translate: (value: number, label: LabelType): string => {
                    return value + '€';
                }
            };
        }
    }

    filter() {
        this.eventEmitterFilter.emit({'ceil': this.max, 'floor': this.min});
    }
}
```

This is the html side:` <ng5-slider  class="custom-slider" [(value)]="min"  [(highValue)]="max" [options]="options" ></ng5-slider> `
This a link for a deployed app <a href="http://165.227.20.253:4200/articles">Online demo</a>, and the part of code source could be found here.https://github.com/ahmed-bhs/angular6-demo/blob/master/src/app/article/price-filter/price-filter.component.ts

**Comment by piotrdz** [comment]
There may be a couple of things going on here:
1. Model values min and max are set before the options object. With recent changes to the slider, this is not likely to work as expected, as the values are verified against current floor and ceil range. See issue #56 for another instance of this. I would change the code to set options object first, then change the min and max model values.

2. min and max are converted to numbers, which is good, but floor and ceil in options object may still be set to strings, depending on what input is provided to the component. This is again not likely to work with current slider code. I find that doubly frustrating because Typescript compiler will silently allow this as conversion from any type. I would convert the input values to numbers and use the number values to set min & max as well as floor & ceil.

**Comment by piotrdz** [comment]
Can you please re-test your code with v1.1.12?

**Comment by ahmed-bhs** [comment]
Hey @piotrdz,
Thank you so mush for all the effort you took to solve the problem. I am very thankful.
Now it's fixed with `ng-slider v1.1.12` the value changes as soon as the response returns a result asynchronously.<br>
For some enhancement, I discovered that I do not need to use min and max and I referred to <a href="https://angular-slider.github.io/ng5-slider/docs">reactive form</a>:

```
    sliderForm: FormGroup;
    options: Options;

    ngOnChanges() {

        this.sliderForm = new FormGroup({
            sliderControl: new FormControl([this.floor, this.ceil])
        });

        if (this.floor && this.ceil) {

            this.options = {
                floor: this.floor,
                ceil: this.ceil,
                showSelectionBar: true,
                selectionBarGradient: {
                    from: 'white',
                    to: '#007bff'
                },
                showOuterSelectionBars: true,
                translate: (value: number, label: LabelType): string => {
                    return value + '€';
                }
            };
        }
    }

```

```
  <form [formGroup]="sliderForm">
      <ng5-slider *ngIf="options" [options]="options" formControlName="sliderControl" (userChangeEnd)="filter()"
      ></ng5-slider>
  </form>
```

**Comment by genyklemberg** [comment]
I had the same issue with the positioning in my dialog window and I fixed this while putting on timer my slider instantiation
